johnlugton/lgtm-issue-tracker-example
# CLI Tool

@johnlugton As mentioned, this splits up the GitHub and JIRA functions into their own modules. It also introduces a basic CLI tool. Here are a few examples:
* Synchronize a GitHub repo and a JIRA project:
```
./gh2jira sync  \
  --gh-url https://api.github.com \
  --gh-user zbazztian \
  --gh-token <ghtoken> \
  --gh-org ghas-performance-testing \
  --gh-repo newrepository \
  --jira-url https://jira.demo-stack.com  \
  --jira-user zbazztian \
  --jira-token <jira-password> \
  --jira-project JIR
```
* install the JIRA webhook:
```
./gh2jira hooks install \ 
  --hook-url https://something.com \ 
  --hook-secret <some-secret> \
  --jira-url https://jira.demo-stack.com \ 
  --jira-user zbazztian \
  --jira-token <jira-password>
```
* install the GitHub webhook for a repository (omit the --gh-repo param if webhook should be for entire organization)
```
./gh2jira hooks install \
  --gh-url https://api.github.com \
  --gh-user zbazztian \
  --gh-token <github-token> \
  --gh-org ghas-performance-testing \
  --gh-repo newrepository
```
* Lists existing webhooks:
```
./gh2jira hooks list  \
  --gh-url https://api.github.com \
  --gh-user zbazztian \
  --gh-token <github-token> \ 
  --gh-org ghas-performance-testing \
  --gh-repo newrepository \
  --jira-url https://jira.demo-stack.com \ 
  --jira-user zbazztian \
  --jira-token <jira-password>
```

It roughly works, but needs a lot more polishing. I will amend this PR tomorrow. Of course, if you are working on it today or are testing it, feel free to amend your improvements.
